k2/csrc/ragged_ops.cu

```diff
@@ -223,6 +223,122 @@ RaggedShape RaggedShape3(Array1<int32_t> *row_splits1,
   return RaggedShape(axes);
 }
 
+
+RaggedShape RaggedShape4(Array1<int32_t> *row_splits1,
+                         Array1<int32_t> *row_ids1, int32_t cached_tot_size1,
+                         Array1<int32_t> *row_splits2,
+                         Array1<int32_t> *row_ids2, int32_t cached_tot_size2,
+                         Array1<int32_t> *row_splits3,
+                         Array1<int32_t> *row_ids3, int32_t cached_tot_size3) {
+  NVTX_RANGE(__func__);
```

Comment:
Just a question: can we implement `RaggedShape4` with `RaggedShape3` and `RaggedShape2` and then `ComposeShape`, it may save a lot of code as `ComposeShape` is very cheap.

Reply:
MM yeah, let me change it in this PR, I'll change RaggedShape3() as well.